src/repr/src/scalar.rs

```diff
@@ -1676,19 +1677,24 @@ impl<'a> ScalarType {
 }
 
 lazy_static! {
-    static ref EMPTY_ARRAY_ROW: Row = unsafe {
-        Row::from_bytes_unchecked(vec![
-            22, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
-        ])
+    static ref EMPTY_ARRAY_ROW: Row = {
```

Comment:
I actually think this is probably more correct, but just pointing out that this is changing the empty array from a 0 length 1 dimensional array to a 0 length 0 dimensional array.

Reply:
Hah, have you developed a mental Row decoder?! PostgreSQL normalizes the empty array representation to use zero dimensions...

https://github.com/MaterializeInc/materialize/blob/9741bf3a7eb4caaa52ec479281f22fe0b7f647b1/src/expr/src/scalar/func.rs#L5121-L5124

...and we obviously are screwing that up in a bunch of places. Apparently it doesn't matter that much? 🤷🏽